src/function/FunctionCreateVars.ts

```diff
@@ -118,6 +119,7 @@ export function addDiagram(
     representation: representation,
     id: diagramID,
     iri: iri ? iri : getNewDiagramContextIRI(diagramID),
+    graph: graph ? graph : getNewDiagramContextIRI(diagramID),
```

Comment:
This should be generating new diagram IRI instead of the context IRI - but I understand you probably have the same logic for both.